pkg/config/remote/client/client.go

```diff
@@ -307,9 +306,34 @@ func (c *Client) startFn() {
 // pollLoop should never be called manually and only be called via the client's `sync.Once`
 // structure in startFn.
 func (c *Client) pollLoop() {
-	successfulFirstRun := false
+	successfulFirstRun := true
+	// First run
+	err := c.update()
+	if err != nil {
+		if status.Code(err) == codes.Unimplemented {
+			// Remote Configuration is disabled as the server isn't initialized
+			//
+			// As this is not a transient error (that would be codes.Unavailable),
+			// stop the client: it shouldn't keep contacting a server that doesn't
+			// exist.
+			log.Debugf("remote configuration isn't enabled, disabling client")
```

Comment:
It's already in place. If it is transient the code would be `Unavailable`, not `Unimplemented`. `Unimplemented` has to be returned explicitly by the server.

Reply:
RC explicitly disabled: `Unimplemented` ([source](https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/grpc.go#L114))
Core agent down: `Unavailable` (from gRPC)

Reply:
Yeah, so my points is still valid. If the core agent is restarted with RC becoming available this change will not make this client work with RC until it's restarted too

Reply:
This is a valid edge case but it already exists today and I don't intend to fix it in this PR. This being said, I don't see situations where it could happen as the RC parameter should be set on every agent process.